src/hotspot/share/opto/loopTransform.cpp

```diff
@@ -2334,7 +2334,13 @@ void PhaseIdealLoop::do_unroll(IdealLoopTree *loop, Node_List &old_new, bool adj
         register_new_node(adj_cmp, ctrl);
         Node* adj_bool = new BoolNode(adj_cmp, bt);
         register_new_node(adj_bool, ctrl);
-        new_limit = new CMoveINode(adj_bool, adj_limit, adj_max, TypeInt::INT);
+        // Prevent type from becoming too pessimistic due to type underflow. The new limit
+        // may be arbitrarily decreased by unrolling, but we know it is in [min_int, hi].
```

Comment:
The comment only applies to stride > 0

Reply:
Thanks for the suggestion, I packed it into the new refactoring/renaming.